core/src/main/java/tc/oc/pgm/spawns/states/Dead.java

```diff
@@ -80,30 +76,36 @@ public void leaveState(List<Event> events) {
     bukkit.removePotionEffect(PotionEffectType.BLINDNESS);
     bukkit.removePotionEffect(PotionEffectType.CONFUSION);
 
+    // If regular rotting didn't end, force-finish it to avoid client-side
+    endRotting();
+
     super.leaveState(events);
   }
 
   @Override
   public void tick() {
     long age = age();
 
-    if (!kitted && ticksUntilRespawn() <= 0) {
+    if (!kitted && ticksUntilRespawn() <= 0 && age >= 20) {
```

Comment:
Maybe dub this internal 20 tick limit for the death kit to some type of "quick respawn" in code?
Something like
```java
private static final long QUICK_RESPAWN_LIMIT = 20
```

So that
```suggestion
    if (!kitted && ticksUntilRespawn() <= 0 && age >= QUICK_RESPAWN_LIMIT) {
```
Or something similar...

Reply:
Reason is I think it would be easier to understand when reading, but its a personal preference

Reply:
Probably leaning more towards `MIN_KIT_TCKS` or similar, it's just to avoid giving you the item to leave/switch team instantly if you die and min respawn time is instant, and there isn't auto respawn. It's quite a stretch, but you never know who may make use of such combination in a map.

Reply:
I like your naming better, and I agree it makes sense to limit giving the kit when the respawn is under one second(ish).